docs/adr/0002-use-mask-as-from-address.md

```diff
@@ -0,0 +1,200 @@
+
+## Pros and Cons of the Options
+
+### Option 1: Keep the Relay Reply Address as the `From:` address
+
+If the sender's email headers are:
+
+```
+Subject: A special offer for you
+From: "Your friends at service X" <offers@servicex.example.com>
+To: r4nd0m@mozmail.com
+```
+
+Relay forwards with:
+
+```
+Subject: A special offer for you
+From: "offers@servicex.example.com [via Relay]" replies@relay.firefox.com
+To: users-real-email@mail.example.com
+Reply-To: replies@relay.firefox.com
+```
+
+- Good, because the sender's email is visible in the mail client (the "display name").
+- Bad, because all Relay emails come from the same address, reducing the information
+  for filtering and processing.
+- Bad, because Proton Mail groups all Relay emails together when in conversation mode.
+
+### Option 2: Use the Service's Sender Address as the `From:` Address
+
+If the sender's email headers are:
+
+```
+Subject: A special offer for you
+From: "Your friends at service X" <offers@servicex.example.com>
+To: r4nd0m@mozmail.com
+```
+
+Relay forwards with:
+
+```
+Subject: A special offer for you
+From: "offers@servicex.example.com [via Relay]" offers@servicex.example.com
+To: users-real-email@mail.example.com
+Reply-To: replies@relay.firefox.com
+Resent-From: r4nd0m@mozmail.com
+Resent-Sender: replies@relay.firefox.com
+```
+
+This format was inspired by the
+[ReSender example](https://datatracker.ietf.org/doc/html/rfc5598#section-5.2) from
+[RFC 5598: Internet Mail Architecture](https://datatracker.ietf.org/doc/html/rfc5598).
+It used standards-based headers to make it clear who sent the message and who
+resent it.
+
+- Good, because the sender's email address is available for filtering.
+- Bad, because [DKIM](https://en.wikipedia.org/wiki/DomainKeys_Identified_Mail)
+  verification caused messages to be marked as spam or dropped entirely.
+
+### Option 3: Use the Relay Mask as the `From:` Address
+
+If the sender's email headers are:
+
+```
+Subject: A special offer for you
+From: "Your friends at service X" <offers@servicex.example.com>
+To: r4nd0m@mozmail.com
+```
+
+Relay forwards with:
+
+```
+Subject: A special offer for you
+From: "offers@servicex.example.com [via Relay]" r4nd0m@mozmail.com
+To: users-real-email@mail.example.com
+Reply-To: replies@relay.firefox.com
+Resent-From: offers@servicex.example.com
+```
+
+- Good, because the original sender is visible in the display name.
+- Good, because the email mask is available for filtering.
+- Good, because Proton Mail organizes by mask in conversation mode.
+- Good, because it rewards usage of a mask on a single service, which is better
+  for user privacy and is compatible with the Relay business model.
+- Bad, because of a small increased probability of delivery to spam, due to the
+  reputation of `mozmail.com` versus `mozilla.com`.
+
+In QA testing in August 2023, emails from one service were sent to spam with
+the new `From:` address, while were delivered to the inbox with the old `From:`
```

Comment:
suggestion (blocking): seems like there's a number missing here? "... while where delievered to the inbox ..."

Reply:
good catch! Changing the paragraph to:

> In QA testing in August 2023, over 100 services had no delivery change with the new `From:` address. For one service, the email went to spam instead of the inbox.